### PR TITLE
[FIX] check for monster before try to back to route

### PR DESCRIPTION
**Pull request by alisonrag**
add checkmonster to ai queue give attackauto time to find other monster beforeback to route again, this avoid openkore to lost main route and walk away from monsters then back to kill

### CODE REVIEW

**Comment by hubertyang88** [comment]
Nice!! Tested for 30 minutes , no funny movement before or after attck both in random or planLockMapWalk mode. 
Now, random walk can keep the previous destination after attck instead of choosing a new one ,  Good!

File: src/AI/CoreLogic.pm [hunk 1]
```diff
@@ -174,6 +174,7 @@ sub iterate {
 	Misc::checkValidity("AI part 3");
 	processAutoEquip();
 	processAutoAttack();
+	processCheckMonster();
```
Comment (c4c1n6kr3m1) [inline src/AI/CoreLogic.pm:177]: i think it's better on MANUAL AI  ???

File: src/AI/CoreLogic.pm [hunk 2]
```diff
@@ -3584,4 +3583,17 @@ sub processPartyShareAuto {
 	}	
 }
 
+sub processCheckMonster {
+	return if AI::inQueue("attack");
+	return if !AI::inQueue("checkMonsters");
```
Comment (c4c1n6kr3m1) [inline src/AI/CoreLogic.pm:3588]: no need for this because next line is AI::is ???

**Comment by onstop** [comment]
capture-20200531-032247
3516  + if (!$field->isCity && !AI::inQueue("storageAuto", "buyAuto") && $config{dcOnPlayer}

**Comment by alisonrag** [comment]
@onstop

**Comment by hubertyang88** [comment]
about timeout , Should there be no "if" , or it will only time out for the first time?

$timeout{'ai_check_monster_auto'}{'time'} = time;

**Pull request merged**